chainer/dataset/tabular/_transform.py

```diff
@@ -0,0 +1,120 @@
+import six
+
+from chainer.dataset.tabular import tabular_dataset
+
+
+class _Transform(tabular_dataset.TabularDataset):
+
+    def __init__(self, dataset, keys, transform):
+        if not isinstance(keys, tuple):
+            keys = keys,
+
+        self._dataset = dataset
+        self._keys = keys
+        self._mode = None
+        self._transform = transform
+
+    def __len__(self):
+        return len(self._dataset)
+
+    @property
+    def keys(self):
+        return self._keys
+
+    @property
+    def mode(self):
+        if self._mode is None:
+            self.get_examples([0], None)
+        return self._mode
+
+    def get_examples(self, indices, key_indices):
+        if key_indices is None:
+            key_indices = six.moves.range(len(self._keys))
+
+        in_examples = self._dataset.get_examples(indices, None)
+
+        out_examples = tuple([] for _ in key_indices)
+        for in_example in six.moves.zip(*in_examples):
+            if self._dataset.mode is tuple:
+                out_example = self._transform(*in_example)
+            elif self._dataset.mode is dict:
+                out_example = self._transform(
+                    **dict(six.moves.zip(self._dataset.keys, in_example)))
+
+            if not isinstance(out_example, (tuple, dict)):
+                out_example = out_example,
+            if isinstance(out_example, tuple):
+                self._mode = tuple
```

Comment:
When `_transform` is implemented incorrectly and it returns an object of different types depending on the example, `self._mode` may be changed over time. Is it intended? If not, how about raising an error in such a case?

Reply:
Current implementation assumes `_transform` is correctly implemented. OK, I'll add a check.